client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -221,5 +232,52 @@ private void validateStreamCut(Map<Stream, StreamCut> streamCuts) {
                }
            });
        }
-   }
+    }
+
+    private static class ReaderGroupConfigSerializer
+            extends VersionedSerializer.WithBuilder<ReaderGroupConfig, ReaderGroupConfigBuilder> {
+        @Override
+        protected ReaderGroupConfigBuilder newBuilder() {
+            return builder();
+        }
+
+        @Override
+        protected byte getWriteVersion() {
+            return 0;
+        }
+
+        @Override
+        protected void declareVersions() {
+            version(0).revision(0, this::write00, this::read00);
+        }
+
+        private void read00(RevisionDataInput revisionDataInput, ReaderGroupConfigBuilder builder) throws IOException {
+            builder.automaticCheckpointIntervalMillis(revisionDataInput.readLong());
+            builder.groupRefreshTimeMillis(revisionDataInput.readLong());
+            ElementDeserializer<Stream> keyDeserializer = in -> Stream.of(in.readUTF());
+            ElementDeserializer<StreamCut> valueDeserializer = in -> StreamCut.fromBytes(ByteBuffer.wrap(in.readArray()));
```

Comment:
`StreamCut.fromBytes` does not take in a byte array?

Reply:
It takes a byte buffer so as to avoid users being forced to make a copy to call our API.

Reply:
I have made this comment before. To be consistent with other Java APIs, we might want to have to/fromByteArray and make `to/fromBytes return and take a byte array.